contrib/pipelines/secadvisor/README.md

```diff
@@ -1,34 +1,85 @@
 # Security Advisor Pipeline
 
-Implements the IBM Security Advisor pipeline
+The IBM Security Advisor pipeline extracts flow data from skydive, filters out
+external traffic, transforms flow records by adding status information, and
+flow classification (egress, ingress, internal), batches flows records together
+into streams (by time or count), then encodes these blocks in JSON form, and
+compresses using GZIP and finally stores as objects into any S3 compatible
+target.
 
+## Deploy Skydive
 
-## Setup Skydive
-
-First build and run skydive:
+Follow instructions on http://skydive.network/documentation/build to install prerequisites and prepare your machine to build the skydive code.
+Build and run skydive:
 
 ```
 make
-cp etc/skydive.yml.default skydive.yml
-sudo $(which skydive)) allineone -c skydive.yml
+sudo $(which skydive) allineone -c etc/skydive.yml.default
 ```
 
 ## Setup Minio
 
-Install and run minio (an AWS S3-like object store).
+For running tests on a local machine, it is possible to set up a local Minio
+object store.
+
+First run the minio server:
 
-## Setup Pipeline
+```
+wget https://dl.min.io/server/minio/release/linux-amd64/minio
+chmod +x minio
+mkdir /tmp/data
+export MINIO_VOLUMES="/var/lib/minio"
+export MINIO_ACCESS_KEY=user
+export MINIO_SECRET_KEY=password
+./minio server /tmp/data
+```
 
-Build and run the pipeline:
+Next setup the bucket:
 
 ```
-make all
-make install
-make run
+wget https://dl.minio.io/client/mc/release/linux-amd64/mc
+chmod +x mc
+./mc config host add local http://localhost:9000 user password /var/lib/minio --api S3v4
+./mc rm --recursive --force local/bucket
+./mc rb --force local/bucket
+./mc mb --ignore-existing local/bucket
 ```
 
-## Generate Flows
+## Run the Pipeline
 
-Via the Skydive WebUI setup captures and generate traffic which should
-result in the secadvisor pipeline sending flows to the ObjectStore.
+Build and run the pipeline in the secadvisor directory:
 
+```
+cd contrib/pipelines/secadvisor
+make
+secadvisor secadvisor.yml.default
+```
+
+## Create Capture
+
+Setup capture on every interface:
+
+```
+skydive client capture create \
+	--gremlin "G.V().Has('Type', 'device', 'Name', 'eth0')" \
+	--type pcap \
+	-c etc/skydive.yml.default
+```
+
+## Inject Traffic
+
+```
+skydive client inject-packet create --count 1000 --type tcp4 \
+	--src "G.V().Has('Type', 'device', 'Name', 'eth0')" \
+	--dst "G.V().Has('Type', 'device', 'Name', 'eth0')" \
+	-c etc/skydive.yml.default
+```
+
+## Verify Operation
+
+Wait for 60 seconds and then check that we have object accumulate in the bucket
+as a result of the operation.
+
+```
+./mc ls --recursive local/bucket
```

Comment:
k, can do that also from the UI

Reply:
how? we want to ensure that flows reached the object store, so it is not enough to see the flows via the webui

Reply:
--> webui of minios .... you can see the objects in minios UI (http://localhost:9000)

Reply:
README is for developers ... I assume they prefer command line (as they can build automation atop it).